sematic/resolvers/cloud_resolver.py

```diff
@@ -78,6 +84,12 @@ def set_graph(self, runs: List[Run], artifacts: List[Artifact], edges: List[Edge
         self._artifacts = {artifact.id: artifact for artifact in artifacts}
         self._edges = {make_edge_key(edge): edge for edge in edges}
 
+    def _get_run(self, run_id) -> Run:
+        # Should refresh from DB for remote exec
+        run = api_client.get_run(run_id)
+        self._runs[run_id] = run
+        return run
```

Comment:
Why are we doing this?

Reply:
again, it's about making sure that before you change a field in `run`, you first get a fresh copy in case it has been changed by the server (due to state refresh, maybe a change to the run made by the UI like the addition of a tag, etc..)